update-version.sh

```diff
@@ -20,6 +20,13 @@ version=$(git describe --tags --abbrev=0)
 version_without_v=$(echo ${version} | cut -f2 -dv)
 echo "${version_without_v}" > VERSION
 
+echo "Updating sub-project versions"
+# First part finds all CMakeList.txt that is 2 spaces deep and no more, limiting it to things like eventstream_rpc/CmakeLists.txt.
+# The second part runs sed (using -exec) and replaces the version string in "set(ENV{_sub_project_version} 1.0.0" with the version of the SDK
+find . -name "CMakeLists.txt" -maxdepth 2 -mindepth 2 -exec sed -i -r -e 's/set(ENV{_sub_project_version} "[^"]*"/set(ENV{_sub_project_version} "'${version_without_v}'"/' {} +
+# This just removes any backup files created by "sed" after calling it.
+find . -name "CMakeLists.txt-r" -maxdepth 2 -mindepth 2 -exec rm {} +
```

Comment:
One of the reasons we went with a `VERSION` file in https://github.com/awslabs/aws-crt-cpp/pull/412 was to avoid the complexity of editing a specific string within a CMakeLists.txt file. It's simpler to read/write a dead-simple file with nothing else in it, than do pattern matching shenanigans like this

Let's do one of these things:
1) Since (none? most?) of the subprojects can't actually be built without building the top-level directory, let's just keep things simple and just assume the top-level project is passing in the version
OR
2) if SIMPLE_VERSION isn't set, read it from "../VERSION" instead of having the value hard-coded here

Reply:
That is fair. I added this because there was discussion on whether these modules can be built outside of the C++ V2 SDK, which I found that some could, though to be honest I doubt anyone does because I cannot see the advantage of doing it that way. Regardless, you can just move some of the sub-projects to a completely different folder and build individually should you so choose. That is why I added the hard-coded string, as a fallback in these cases.

However, I adjusted this using a combination of both things you suggested. Now it checks to see if `SIMPLE_VERSION` exists and if it does, then it uses that, otherwise it checks for the `../VERSION` file to see if it exists and uses it directly if it does. If both of these checks fail, then it just does not version stamp it and prints the version (in the console) out as `unknown`.

This avoids the need to do any pattern matching and still properly version stamps everything in the SDK.